lib/ide/src/controller/module.rs

```diff
@@ -20,10 +21,64 @@ use data::text::TextChangedNotification;
 use file_manager_client as fmc;
 use json_rpc::error::RpcError;
 use parser::api::IsParser;
+use parser::api::ModuleWithMetadata;
 use parser::Parser;
+
+use serde::Serialize;
+use serde::Deserialize;
 use shapely::shared;
 
 
+
+// ============
+// == Module ==
+// ============
+
+/// Parsed file / module with metadata
+#[derive(Debug,Clone)]
+pub struct Module {
+    /// ast representation
+    pub ast: Ast,
+    /// strongly typed metadata
+    pub metadata: Metadata
+}
+
+impl TryFrom<ModuleWithMetadata> for Module {
+    type Error = serde_json::Error;
+
+    fn try_from(value:ModuleWithMetadata) -> Result<Self,Self::Error> {
+        let metadata = Metadata::deserialize(value.metadata)?;
+        Ok(Module {ast:value.ast,metadata})
+    }
+}
+
+impl From<Module> for ModuleWithMetadata {
+    fn from(value:Module) -> Self {
+        let metadata = serde_json::to_value(value.metadata).expect(
+            "Should be possible to serialize metadata to json."
+        );
+        ModuleWithMetadata { ast:value.ast, metadata }
```

Comment:
Sometimes you put spces after `{` and before `}` , sometimes not. The correct way is not to put them there.